operator/validator/event_handler.go

```diff
@@ -198,30 +201,46 @@ func (c *controller) handleValidatorAddedEvent(
 		}
 	}
 
-	validatorShare, found, err := c.sharesStorage.GetShare(event.PublicKey)
-	if err != nil {
-		return nil, errors.Wrap(err, "could not check if validator share exist")
+	validatorShare, found, shareErr := c.sharesStorage.GetShare(event.PublicKey)
+	if shareErr != nil {
+		return nil, errors.Wrap(shareErr, "could not check if validator share exist")
+	}
+	// Prevent multiple registration of the same validator with different owner address
+	// owner A registers validator with public key X (OK)
+	// owner B registers validator with public key X (NOT OK)
+	if found && event.Owner != validatorShare.OwnerAddress {
+		err = &abiparser.MalformedEventError{
+			Err: errors.Errorf(
+				"validator share already exists with different owner address: expected %s, got %s",
+				validatorShare.OwnerAddress.String(),
+				event.Owner.String(),
+			),
+		}
+		return nil, err
 	}
 	if !found {
```

Comment:
I assume `found && event.Owner == validatorShare.Owner` is prevented on the contract level?

Reply:
Right, it's handled on the contract side